ui-admin/src/study/consents/CreateConsentModal.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useContext, useState } from 'react'
+import { StudyEnvContextT } from '../StudyEnvironmentRouter'
+import { ConsentForm } from '@juniper/ui-core'
+import { PortalContext, PortalContextT } from 'portal/PortalProvider'
+import { useNavigate } from 'react-router-dom'
+import Api from 'api/api'
+import { Store } from 'react-notifications-component'
+import { successNotification } from 'util/notifications'
+import Modal from 'react-bootstrap/Modal'
+import InfoPopup from 'components/forms/InfoPopup'
+import LoadingSpinner from 'util/LoadingSpinner'
+import { doApiLoad } from 'api/api-utils'
+import { useFormCreationNameFields } from '../surveys/CreateSurveyModal'
+
+/** creates a new consent form for a study */
+export default function CreateConsentModal({ studyEnvContext, onDismiss }: {
+    studyEnvContext: StudyEnvContextT
+    onDismiss: () => void }) {
+  const [isLoading, setIsLoading] = useState(false)
+  const { formName, formStableId, clearFields, nameInput, stableIdInput } = useFormCreationNameFields()
+
+  const portalContext = useContext(PortalContext) as PortalContextT
+  const navigate = useNavigate()
+
+
+  const attachConsentToEnv = async (createdConsent: ConsentForm) => {
```

Comment:
Wondering why these two operations are not combined into one on the backend?
It feels like even if there were instances where a consent form would be created and attached to different entities, the backend would handle that, perhaps through parameters that indicate what is desired?

Reply:
that was how this API was originally structured, the backend took a "targetStudyEnvs" property.  But we want the form save to succeed no matter what, since that's saving user work.  And so it led to complicated error status/message passing for the backend to be able to communicate "the form save succeeded but it couldn't be attached".  It wasn't a big deal, and I think both approaches are totally valid, but IMO this approach was slightly more readable on both the front and backend.